pkg/apis/provisioning/v1alpha5/constraints.go

```diff
@@ -33,7 +33,7 @@ type Constraints struct {
 	// +optional
 	Taints Taints `json:"taints,omitempty"`
 	// Requirements are layered with Labels and applied to every node.
-	Requirements Requirements `json:"requirements,omitempty"`
+	Requirements *Requirements `json:"requirements,inline,omitempty"`
```

Comment:
```suggestion
	Requirements *Requirements `json:",inline"`
```

Reply:
Tried and it doesn't work without those two tags.
Without requirements, there is an extra `Requirements: []v1.NodeSelectorRequirement` and any future changes to the requirement objects appears under key `Requirements`.
The omitempty is needed in case for Provisioner.Spec without requirements. Without it, there will be nil pointer errors.

Reply:
Can we make it a non-pointer? Let's pair program it.

Reply:
Yeah I thought about it. Then I realize initialization will become painful.